src/adapters/adgear.js

```diff
@@ -0,0 +1,130 @@
+  var baseAdapter = Adapter.createNew('adgear');
+
+  baseAdapter.callBids = function(params) {
+    var bidderCode = baseAdapter.getBidderCode();
+    var bids = [];
+
+    utils._each(params.bids, function(bid) {
+      if (bid.placementCode && bid.sizes && bid.params.tagId) {
+
+        /* We allow overriding 'sizes' via bid.params so check there first: */
+        var sizes = bid.params.hasOwnProperty('sizes') ? bid.params.sizes : bid.sizes;
+        sizes = utils.parseSizesInput(sizes);
+
+        /* We allow publisher overriding 'url' via bid.params so check there first: */
+        var durl = bid.params.hasOwnProperty('url') ? bid.params.url : window.location.href;
+
+        utils._each(sizes, function(format) {
+          var url = htag + bid.params.tagId + '.js?';
+          url = utils.tryAppendQueryString(url, 'AG_CB', 'pbjs.handleAdGearCallback');
+          url = utils.tryAppendQueryString(url, 'AG_SZ', encodeURIComponent(format));
+          url = utils.tryAppendQueryString(url, 'AG_TZ', (new Date()).getTimezoneOffset());
+          url = utils.tryAppendQueryString(url, 'AG_URL', encodeURIComponent(durl));
+          url = utils.tryAppendQueryString(url, 'AG_R', String(Math.floor(Math.random() * 1000000000)));
+
+          var cuid = utils.getUniqueIdentifierStr();
+          url = utils.tryAppendQueryString(url, 'AG_CBID', encodeURIComponent(cuid));
+          bidmanager.pbCallbackMap[cuid] = {
+            'bidObj': bid,
+            'url': url,
+            'start': new Date().getTime()
+          };
+          bids.push(bidmanager.pbCallbackMap[cuid]);
+        });
+
+      }
+    });
+
+    bidmanager.setExpectedBidsCount(bidderCode, bids.length);
+
+    utils._each(bids, function(bid) {
+      utils.logMessage('[AdGear] Dispatching header bidrequest to URL ' + bid.url);
+      adloader.loadScript(bid.url);
+    });
+  };
+
+  /*
+   * AdGear's htag bidResponse should look like:
+   *
+   * {
+   *   'cbid': 'uniqueStashId',  // AG_CBID value passed into handler
+   *   'wincur': 'USD',          // or CAD
+   *   'wincpm': 1.99,           // the win price as a Number, in dollars
+   *   'dealid': null,           // or string value of winning deal ID
+   *   'width': 300,             // width in pixels of winning ad
+   *   'height': 250,            // height in pixels of winning ad
+   *   'html': 'tag_here'        // HTML tag to load if we are picked
+   * }
+   *
+   * If we're given back a blank/nobid, then wincur, wincpm, dealid, width, height, and html
+   * will be null or absent from the bidResponse. On the other hand, a default tag will come
+   * back with these non-null and wincpm = 0
+   */
+  pbjs.handleAdGearCallback = function(bidResponse) {
+    if (bidResponse && bidResponse.cbid) {
+      var bidStash = bidmanager.getPlacementIdByCBIdentifer(bidResponse.cbid);
+      var bid = null;
+
+      if (bidStash) {
+
+        var bidObj = bidStash.bidObj;
+        bidObj.status = CONSTANTS.STATUS.GOOD;
```

Comment:
This line isn't required anymore. You can use it for creating the bid like so: `bid = bidfactory.createBid(CONSTANTS.STATUS.GOOD);`